src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -196,13 +238,23 @@ private async Task ProcessQueueAsync()
                         // will be sent back to the client but they can also be captured via HandleNonMutatingRequestError,
                         // though these errors don't put us into a bad state as far as the rest of the queue goes.
                         // Furthermore we use Task.Run here to protect ourselves against synchronous execution of work
-                        // blocking the request queue for longer periods of time (it enforces parallelizabilty).
-                        _ = WrapStartRequestTaskAsync(Task.Run(() => work.StartRequestAsync(context, cancellationToken), cancellationToken), rethrowExceptions: false);
+                        // blocking the request queue for longer periods of time (it enforces parallelizability).
+                        var pendingTask = WrapStartRequestTaskAsync(Task.Run(() => work.StartRequestAsync(context, cancellationToken), cancellationToken), rethrowExceptions: false);
```

Comment:
Can we avoid putting things into the `pendingTasks` dictionary at all if `CancelInProgressWorkUponMutatingRequest` is false? It would save us 2 `ConcurrentDictionary` operations and a `ContinueWith` on every non-mutating request which seems like good value.

Reply:
Seems reasonable. Let me see what it looks like.